### PR TITLE
Blog spike

### PR DESCRIPTION
**Pull request by myronmarston**
This is the a spike to start getting the blog going.  I ported over a simple blog post from my blog (http://myronmars.to/n/dev-blog/2014/06/rspec-team-changes) to see how it would look here.  I also tried porting a more complicated post (with snippets and everything) but it rendered kinda oddly.

BTW, here's a non-exhaustive list of TODOs before this is fully ready:

- [x] Some kind of blog index/landing page at `/blog`. (being worked on by @xaviershay)
- [x] Header link to `/blog`. (being worked on by @xaviershay)
- [x] Full styles for all the typical elements used in blog posts (code snippets, headers, subheaders, lists, etc).  I tried porting over the markdown file that's the jekyll source of http://myronmars.to/n/dev-blog/2014/09/rspec-3-1-has-been-released and it didn't look good at all. (Some of that is probably the use of liquid tags in the markdown file, which jekyll allows, so it may mostly be a matter of cleaning up the source). (being worked on by @xaviershay)
- [x] Comment support via disqus. (being worked on by @myronmarston)
- [x] Ensure the XML feed works. (being worked on by @myronmarston)
- [x] In 2175d35e3b87ea03626dfb8d1a0b1917399a0ca2 I generated a bunch of files from `middleman init . --template=blog` and I think we probably won't be using a bunch of that, so it'd be nice to clean it up.
- [ ] The generated config contains lots of settings I don't yet understand.  We should figure out what they do and what settings we want to use.  There may be features of middleman's blogging engine that we'll want to leverage or disable.
- [x] Should we "seed" the blog archive history by migrating over the RSpec blog posts from my blog from the last couple years?  If we did that, I'd setup redirects so that the URLs at my site continue to work.  One issue there is disqus comments: many of those blog posts have extensive comment threads that we'd either lose or have to find a way to migrate over.

/cc @JonRowe

### CODE REVIEW

**Comment by myronmarston** [comment]
We also need to add the publication date to the markup somehow...

**Comment by JonRowe** [comment]
> Blog post announcing the launch of the new site, publicly thanking the folks who have helped us out, and inviting the community to keep helping us improve it further. @JonRowe, do you mind working on that? You know more about who from Ninefold helped with what.

Dev wise @mootpointer and @sjahandideh; Design wise @8apixel (not Ninefold)

**Comment by myronmarston** [comment]
@JonRowe does that mean you'd prefer I write up a blog post about the new website?

**Comment by JonRowe** [comment]
I might try this week, not sure, personal circumstances at the moment.

**Comment by myronmarston** [comment]
OK, I imported the blog posts from my blog.

@JonRowe -- holler if you're able and willing to work on the new website blog post; otherwise I'm planning to try to take a stab at it this weekend with the hope to ship it next week and release 3.2, etc.

**Comment by myronmarston** [comment]
OK, @JonRowe, I took a shot at putting together a blog post on the new site.  Can you take a look?  In particularly I wasn't 100% about how did the logo, so can you check what I said for accuracy?

File: source/blog/2015-01-30-new-rspec-website.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+I'd like to thank all the people who helped make this site happen:
+
+* [Ninefold](https://ninefold.com/), who sponsored the development of
+  the site and are covering its on-going costs.
+* [Katie Chen](http://pistachiomade.com/), who came up with the logo and
+  new design.
+* [Andrew Harvey](http://mootpointer.com/),
+  [Shamim Jahandideh](https://github.com/sjahandideh), and
+  [Jon Rowe](http://jonrowe.co.uk/), who who helped develop the site.
```
Comment (JonRowe) [inline source/blog/2015-01-30-new-rspec-website.md]: `s/who who/who`

File: source/blog/2015-01-30-new-rspec-website.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+title: New RSpec Website!
+author: Myron Marston
+published: false
+---
+
+RSpec's website has been in need of a refresh for a long time. We've
+been working on this new site, off an on, for more than a year, and
```
Comment (JonRowe) [inline source/blog/2015-01-30-new-rspec-website.md]: `s/an/and`

File: source/blog/2015-01-30-new-rspec-website.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+we're glad to finally ship it. We're hoping that it'll prove much
```
Comment (JonRowe) [inline source/blog/2015-01-30-new-rspec-website.md]: repetition of "we're" is a bit weird

File: source/blog/2015-01-30-new-rspec-website.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+* [Ninefold](https://ninefold.com/), who sponsored the development of
+  the site and are covering its on-going costs.
+* [Katie Chen](http://pistachiomade.com/), who came up with the logo and
+  new design.
+* [Andrew Harvey](http://mootpointer.com/),
+  [Shamim Jahandideh](https://github.com/sjahandideh), and
+  [Jon Rowe](http://jonrowe.co.uk/), who who helped develop the site.
+* [Carlos Rodriguez](http://cohitre.com/), who helped out with the CSS.
+
+While what we have no is a huge improvement, it's still not as polished
```
Comment (JonRowe) [inline source/blog/2015-01-30-new-rspec-website.md]: `s/no/now`

File: source/blog/2015-01-30-new-rspec-website.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+as we would like, and we're hoping folks from the community will chip in
+and help us iteratively improve it. The blog in particular--which I
```
Comment (JonRowe) [inline source/blog/2015-01-30-new-rspec-website.md]: spacing on the -- here?

**Comment by JonRowe** [comment]
I believe that's accurate @myronmarston (ping @mootpointer)

**Comment by myronmarston** [comment]
Thanks for the feedback, @JonRowe.  I think I addressed all your comments.

**Comment by mootpointer** [comment]
Looks all good to me. Might want to check with @8apixel (Katie Chen) how she wants to be credited. 


A.

**Comment by myronmarston** [comment]
Good idea.  @8apixel -- we're getting ready to (finally) ship the new RSpec website.  Please take a look at the announcement/thanks blog post (https://github.com/rspec/rspec.github.io/blob/c012a65cb2517c850c349121c741c853249c78c5/source/blog/2015-01-30-new-rspec-website.md) and let us know if that's how you'd like to be credited.  We're happy to change that however you like.

**Comment by myronmarston** [comment]
Merging to get this in master.

**Comment by JonRowe** [comment]
I published the latest copy of `source` to `master` so it's up on rspec.github.io for people to check out :)

**Comment by JonRowe** [comment]
One thing I'm noting is that article headers look a bit out of place on the index, not a lot delinearating articles.

**Comment by myronmarston** [comment]
> One thing I'm noting is that article headers look a bit out of place on the index, not a lot delinearating articles.

I agree.  In the short term I'm thinking of changing our blog index page to just a list of articles (w/o any snippet), like http://myronmars.to/n/dev-blog

**Comment by JonRowe** [comment]
Does the middleman blog extension allow us to specify the abstract? I know jekyll does...

**Comment by myronmarston** [comment]
> Does the middleman blog extension allow us to specify the abstract? I know jekyll does...

https://middlemanapp.com/basics/blogging/#article-summaries

**Comment by JonRowe** [comment]
So we could use the proc option and look for the relevant front matter snippet...

**Comment by myronmarston** [comment]
Sounds like a lot of work, particularly for all the old posts I've imported.  If you want to put the effort in to doing that I'm not going to complain, of course, but I don't personally feel like it's worth the effort.

**Comment by mootpointer** [comment]
What's wrong with the `READMORE` method?

**Comment by JonRowe** [comment]
Doesn't allow you to have a different abstract to the top of the article.

**Comment by mootpointer** [comment]
Ah I see. I can knock out the proc version pretty quickly, being familiar with Middleman.




Maybe start by cutting it back to titles, then I’ll work on a PR over the weekend?

**Pull request merged**